include/libtorrent/bencode.hpp

```diff
@@ -86,9 +86,12 @@ namespace libtorrent
 
 	namespace detail
 	{
-		template <class OutIt>
-		int write_integer(OutIt& out, entry::integer_type val)
+		template <class OutIt, class In, typename Cond
+			= typename std::enable_if<std::is_integral<In>::value>::type>
+		int write_integer(OutIt& out, In data)
 		{
+			entry::integer_type const val(data);
```

Comment:
looking at this again. it feels like there should be something like this: ``static_assert(sizeof(data) <= 8, "no more than 64 bits supported");``. Maybe it's a bit paranoid though.